### DISCUSSION TITLE
Run from Visual Studio Error

### DISCUSSION THREAD

**Discussion by MCVenor** [stride3d/stride#1940] [Newcomers]
Hi,

i totally forgot that you can run the project from Visual Studio, because i tried it a couple of times and got errors. So i thought this is just the way it is. But now i know that the VS Start should work and be used.

My Visual Studio 2022 doesnt run when i try it, what could be the error?

I have tried my current project and a new one, just created, empty project with the shiny ball in the middle.
Both projects show the same behaviour:
- can run them just fine from the Game Studio
- can attach to both
- can't  run from Visual Studio with f5

Clicked on reinstall for the Expansion and VS told me that its already installed.

<details>
<summary> Here is the log from VS from my Project </summary>

Error		Failed to run assembly processor with parameters: --auto-notify-property --parameter-key --auto-module-initializer --serialization --add-reference="C:\Users\Michi-Pc\Documents\Stride Projects\MyGame_Kapusta\MyGame_Kapusta\bin\Debug\net6.0-windows\MyGame_Kapusta.dll" --add-reference="C:\Users\Michi-Pc\Documents\Stride Projects\MyGame_Kapusta\MyGame_Kapusta\bin\Debug\net6.0-windows\MyGame_Kapusta.dll" --references-file="obj\Debug\win-x64\StrideReferences.cache" --platform=Windows --targetFramework= "obj\Debug\win-x64\MyGame_Kapusta.Windows.dll"	MyGame_Kapusta.Windows	C:\Users\Michi-Pc\.nuget\packages\stride.core\4.1.0.1948\buildTransitive\Stride.Core.targets	94	
Error		Check the previous logs	MyGame_Kapusta	C:\Users\Michi-Pc\.nuget\packages\stride.core\4.1.0.1948\buildTransitive\Stride.Core.targets	94	
Error		Check the previous logs	MyGame_Kapusta.Windows	C:\Users\Michi-Pc\.nuget\packages\stride.core\4.1.0.1948\buildTransitive\Stride.Core.targets	94	
Error		Failed to run assembly processor with parameters: --auto-notify-property --parameter-key --auto-module-initializer --serialization --add-reference="C:\Users\Michi-Pc\.nuget\packages\stride.engine\4.1.0.1948\lib\net6.0\Stride.Engine.dll" --add-reference="C:\Users\Michi-Pc\.nuget\packages\stride.video\4.1.0.1948\lib\net6.0\Stride.Video.dll" --add-reference="C:\Users\Michi-Pc\.nuget\packages\stride.physics\4.1.0.1948\lib\net6.0\BulletSharp.NetStandard.dll" --add-reference="C:\Users\Michi-Pc\.nuget\packages\stride.physics\4.1.0.1948\lib\net6.0\Stride.Physics.dll" --add-reference="C:\Users\Michi-Pc\.nuget\packages\stride.navigation\4.1.0.1948\lib\net6.0\Stride.Navigation.dll" --add-reference="C:\Users\Michi-Pc\.nuget\packages\stride.particles\4.1.0.1948\lib\net6.0\Stride.Particles.dll" --add-reference="C:\Users\Michi-Pc\.nuget\packages\stride.ui\4.1.0.1948\lib\net6.0\Stride.UI.dll" --references-file="obj\Debug\net6.0-windows\StrideReferences.cache" --platform=Windows --targetFramework= "obj\Debug\net6.0-windows\MyGame_Kapusta.dll"	MyGame_Kapusta	C:\Users\Michi-Pc\.nuget\packages\stride.core\4.1.0.1948\buildTransitive\Stride.Core.targets	94	

</details>

<details>
<summary>Here is the log from VS from new empty Project </summary>
Error		Check the previous logs	MyNewerGame_TestRunFromVS.Windows	C:\Users\Michi-Pc\.nuget\packages\stride.core\4.1.0.1948\buildTransitive\Stride.Core.targets	94	
Error		Failed to run assembly processor with parameters: --auto-notify-property --parameter-key --auto-module-initializer --serialization --add-reference="C:\Users\Michi-Pc\Documents\Stride Projects\MyNewerGame_TestRunFromVS\MyNewerGame_TestRunFromVS\bin\Debug\net6.0-windows\MyNewerGame_TestRunFromVS.dll" --add-reference="C:\Users\Michi-Pc\Documents\Stride Projects\MyNewerGame_TestRunFromVS\MyNewerGame_TestRunFromVS\bin\Debug\net6.0-windows\MyNewerGame_TestRunFromVS.dll" --references-file="obj\Debug\win-x64\StrideReferences.cache" --platform=Windows --targetFramework= "obj\Debug\win-x64\MyNewerGame_TestRunFromVS.Windows.dll"	MyNewerGame_TestRunFromVS.Windows	C:\Users\Michi-Pc\.nuget\packages\stride.core\4.1.0.1948\buildTransitive\Stride.Core.targets	94	
</details>

**Comment by MCVenor** [comment]
Deleting the projects bin and obj folders, followed by a Solution Rebuild, followed by a Game Assemblies reload gave me this output in the Game Studio. Could this be connected?

<details>
<summary> Error from Game Studio</summary>


 [(0,0)]: Info: Patch for assembly [MyGame_Kapusta, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]
[(0,0)]: Info: MyGame_Kapusta -> C:\Users\Michi-Pc\Documents\Stride Projects\MyGame_Kapusta\MyGame_Kapusta\bin\Debug\net6.0-windows\MyGame_Kapusta.dll
Info: Unloading assembly MyGame_Kapusta, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
Info: Loading assembly 


Error: Error loading assembly MyGame_Kapusta: 
ArgumentNullException: Value cannot be null. (Parameter 'assembly')
   at Stride.Core.Reflection.AssemblyRegistry.Register(Assembly assembly, IEnumerable`1 categories) in C:\BuildAgent\work\b5f46e3c4829a09e\sources\core\Stride.Core\Reflection\AssemblyRegistry.cs:line 189
   at Stride.Assets.Presentation.AssemblyReloading.ReloadAssembliesOperation.LoadAssemblies(ILogger log, AssemblyContainer assemblyContainer, List`1 loadedAssemblies, Boolean newVersion, Boolean firstTime) in C:\BuildAgent\work\b5f46e3c4829a09e\sources\editor\Stride.Assets.Presentation\AssemblyReloading\ReloadAssembliesOperation.cs:line 93

Info: Updating components with newly loaded assemblies

Warning: Could not deserialize object of type 'MyGame_Kapusta.UiEditorDemo'; replacing it with an object implementing IUnloadable
YamlException: Unable to resolve tag [!MyGame_Kapusta.UiEditorDemo,MyGame_Kapusta] to type from tag resolution or registered assemblies ((Lin: 0, Col: 0, Chr: 0) -> (Lin: 0, Col: 0, Chr: 0))
   at Stride.Core.Yaml.Serialization.Serializers.TagTypeSerializer.ReadYaml(ObjectContext& objectContext) in C:\BuildAgent\work\b5f46e3c4829a09e\sources\core\Stride.Core.Yaml\Serialization\Serializers\TagTypeSerializer.cs:line 131
   at Stride.Core.Yaml.ErrorRecoverySerializer.ReadYaml(ObjectContext& objectContext) in C:\BuildAgent\work\b5f46e3c4829a09e\sources\assets\Stride.Core.Assets\Yaml\ErrorRecoverySerializer.cs:line 21

Warning: Could not deserialize object of type 'MyGame_Kapusta.bob.ClickChecker'; replacing it with an object implementing IUnloadable
YamlException: Unable to resolve tag [!MyGame_Kapusta.bob.ClickChecker,MyGame_Kapusta] to type from tag resolution or registered assemblies ((Lin: 0, Col: 0, Chr: 0) -> (Lin: 0, Col: 0, Chr: 0))
   at Stride.Core.Yaml.Serialization.Serializers.TagTypeSerializer.ReadYaml(ObjectContext& objectContext) in C:\BuildAgent\work\b5f46e3c4829a09e\sources\core\Stride.Core.Yaml\Serialization\Serializers\TagTypeSerializer.cs:line 131
   at Stride.Core.Yaml.ErrorRecoverySerializer.ReadYaml(ObjectContext& objectContext) in C:\BuildAgent\work\b5f46e3c4829a09e\sources\assets\Stride.Core.Assets\Yaml\ErrorRecoverySerializer.cs:line 21

</details>


Edit: Just opening my project now gives me this
![grafik](https://github.com/stride3d/stride/assets/12583072/7c4d64bb-8ad0-4c11-984f-fe3a6fa79b9a)


<details>
<summary> Error as Text</summary>

Verbose: Process dependencies for MyGame_Kapusta.Windows...
Verbose: Process dependencies for MyGame_Kapusta...
Verbose: Restore NuGet packages for MyGame_Kapusta...
Verbose: Restore NuGet packages for MyGame_Kapusta.Windows...
Verbose: Loading Assets from Package [Stride.Engine.sdpkg]
Verbose: Loading Assets from Package [Stride.Graphics.sdpkg]
Verbose: Loading Assets from Package [Stride.Particles.sdpkg]
Verbose: Loading Assets from Package [Stride.Rendering.sdpkg]
Verbose: Loading Assets from Package [Stride.SpriteStudio.Offline.sdpkg]
Verbose: Loading Assets from Package [Stride.Video.sdpkg]
Error: Error while loading assembly reference [C:\Users\Michi-Pc\Documents\Stride Projects\MyGame_Kapusta\MyGame_Kapusta\bin\Debug\net6.0-windows\MyGame_Kapusta.dll]
BadImageFormatException: Duplicate type with name 'Stride.Core.DataSerializers.MyGame_Kapusta_BasicCameraControllerSerializer' in assembly 'MyGame_Kapusta, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromPath(IntPtr ptrNativeAssemblyLoadContext, String ilPath, String niPath, ObjectHandleOnStack retAssembly)
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromAssemblyPath(String assemblyPath)
   at System.Reflection.Assembly.LoadFrom(String assemblyFile)
   at Stride.Core.Reflection.AssemblyContainer.LoadAssemblyFromPathInternal(String assemblyFullPath) in C:\BuildAgent\work\b5f46e3c4829a09e\sources\core\Stride.Core.Design\Reflection\AssemblyContainer.cs:line 321

Error: Unable to load assembly reference [C:\Users\Michi-Pc\Documents\Stride Projects\MyGame_Kapusta\MyGame_Kapusta\bin\Debug\net6.0-windows\MyGame_Kapusta.dll]
Verbose: Loading Assets from Package [MyGame_Kapusta.sdpkg]
Warning: Could not deserialize object of type 'MyGame_Kapusta.bob.ClickChecker'; replacing it with an object implementing IUnloadable
YamlException: Unable to resolve tag [!MyGame_Kapusta.bob.ClickChecker,MyGame_Kapusta] to type from tag resolution or registered assemblies ((Lin: 53, Col: 54, Chr: 2774) -> (Lin: 54, Col: 24, Chr: 2847))
   at Stride.Core.Yaml.Serialization.Serializers.TagTypeSerializer.ReadYaml(ObjectContext& objectContext) in C:\BuildAgent\work\b5f46e3c4829a09e\sources\core\Stride.Core.Yaml\Serialization\Serializers\TagTypeSerializer.cs:line 81
   at Stride.Core.Yaml.Serialization.Serializers.ChainedSerializer.ReadYaml(ObjectContext& objectContext) in C:\BuildAgent\work\b5f46e3c4829a09e\sources\core\Stride.Core.Yaml\Serialization\Serializers\ChainedSerializer.cs:line 109
   at Stride.Core.Yaml.ErrorRecoverySerializer.ReadYaml(ObjectContext& objectContext) in C:\BuildAgent\work\b5f46e3c4829a09e\sources\assets\Stride.Core.Assets\Yaml\ErrorRecoverySerializer.cs:line 21

Warning: Could not deserialize object of type 'MyGame_Kapusta.UiEditorDemo'; replacing it with an object implementing IUnloadable
YamlException: Unable to resolve tag [!MyGame_Kapusta.UiEditorDemo,MyGame_Kapusta] to type from tag resolution or registered assemblies ((Lin: 96, Col: 54, Chr: 5121) -> (Lin: 97, Col: 24, Chr: 5190))
   at Stride.Core.Yaml.Serialization.Serializers.TagTypeSerializer.ReadYaml(ObjectContext& objectContext) in C:\BuildAgent\work\b5f46e3c4829a09e\sources\core\Stride.Core.Yaml\Serialization\Serializers\TagTypeSerializer.cs:line 81
   at Stride.Core.Yaml.Serialization.Serializers.ChainedSerializer.ReadYaml(ObjectContext& objectContext) in C:\BuildAgent\work\b5f46e3c4829a09e\sources\core\Stride.Core.Yaml\Serialization\Serializers\ChainedSerializer.cs:line 109
   at Stride.Core.Yaml.ErrorRecoverySerializer.ReadYaml(ObjectContext& objectContext) in C:\BuildAgent\work\b5f46e3c4829a09e\sources\assets\Stride.Core.Assets\Yaml\ErrorRecoverySerializer.cs:line 21

Verbose: Loading Assets from Package [MyGame_Kapusta.Windows.sdpkg]

</details>


EDIT 2:
I managed to narrow it futher down.
A new blank project will compile and run, from Visual Studio and from Game Studio.
But if i add a single new cs file (like Class1.cs) the compilation will no longer work in Visual Studio.

**Reply by Eideren** [reply]
Can't repro on latest master, which commit did you build it off of ?
> But if i add a single new cs file (like Class1.cs) the compilation will no longer work in Visual Studio.

I'm guessing you meant "no longer works in the editor" ?
If you build the game project from VS, what's the build output like ?

**Reply by MCVenor** [reply]
i deleted stride.

so i can download the source and compile it to see what goes wrong exactly.
But no, with "But if i add a single new cs file (like Class1.cs) the compilation will no longer work in Visual Studio."
i mean that if the project is stright from the templates, like no changes, and i open visual studio throu the GameStudios button. If i then f5, it actually compiles and runs the game.
But if i right click on the solution/project in VS and add a cs file throu VS, it no longer compiles. It tries to and stops with Errors.
i will reinstall stried in a bit to reproduce this and report back the error message( iirc it was the inital error message from main post).

The Version was: 
Release Type: Official Release
Version: 4.1.0.1948 (.NET 6.0.11)

**Comment by MCVenor** [comment]
i can not reproduce it, no matter what i try. The Solution runs without errors.

There have been way too many changes, so i cant even name what the fix could be. But maybe this can help someone, here is an list of what i did:

- delete stride ( not stride launcher )
- remove the Stride Extension
- update Visual Studio 22
- without actually building stride, follow these steps from [Build Instructions](https://github.com/stride3d/stride#building-from-source) 
- i installed DotNet SDK 7
- somehow **DotNet SDK 6** got removed in the process and i reinstalled it too, you can check with the cmd "dotnet --info" on windows.
- reinstall stride and stride extension for VS
- new project and then run from VS
